app/src/androidTest/java/com/github/epfl/meili/forum/PostActivityAndroidTest.kt

```diff
@@ -0,0 +1,202 @@
+
+        FirebasePostService.dbProvider = { mockFirestore }
+    }
+
+    @Before
+    fun initiateAuthAndService() {
+        UiThreadStatement.runOnUiThread {
+            //Injecting authentication Service
+            val mockAuthService = MockAuthenticationService()
+            Auth.setAuthenticationService(mockAuthService)
+
+            Auth.isLoggedIn.value = true
+            Auth.email = TEST_EMAIL
+            Auth.name = TEST_USERNAME
+        }
+    }
+
+    @Before
+    fun initIntents() {
+        Intents.init()
+    }
+
+    @After
+    fun releaseIntents() {
+        Intents.release()
+    }
+
+
+
+    @Test
+    fun postActivityAndroidTest() {
+        val materialButton = onView(
+            allOf(
+                withId(R.id.launchForumView), withText("Launch Forum View"),
+                childAtPosition(
+                    childAtPosition(
+                        withId(android.R.id.content),
+                        0
+                    ),
+                    2
+                ),
+                isDisplayed()
+            )
+        )
+        materialButton.perform(click())
+
+        val materialButton2 = onView(
+            allOf(
+                withId(R.id.forum_new_post_button), withText("+"),
+                childAtPosition(
+                    childAtPosition(
+                        withId(android.R.id.content),
+                        0
+                    ),
+                    1
+                ),
+                isDisplayed()
+            )
+        )
+        materialButton2.perform(click())
+
+
+        val appCompatEditText = onView(
+            allOf(
+                withId(R.id.new_post_title),
+                childAtPosition(
+                    childAtPosition(
+                        withId(android.R.id.content),
+                        0
+                    ),
+                    0
+                ),
+                isDisplayed()
+            )
+        )
+        appCompatEditText.perform(replaceText("TEST_TITLE"), closeSoftKeyboard())
+
+        val appCompatEditText2 = onView(
+            allOf(
+                withId(R.id.new_post_text),
+                childAtPosition(
+                    childAtPosition(
+                        withId(android.R.id.content),
+                        0
+                    ),
+                    1
+                ),
+                isDisplayed()
+            )
+        )
+        appCompatEditText2.perform(replaceText("TEST_TEXT"), closeSoftKeyboard())
+
+
+        val materialButton5 = onView(
+            allOf(
+                withId(R.id.new_post_create_button), withText("Create post"),
+                childAtPosition(
+                    childAtPosition(
+                        withId(android.R.id.content),
+                        0
+                    ),
+                    2
+                ),
+                isDisplayed()
+            )
+        )
+        materialButton5.perform(click())
+    }
```

Comment:
Yeah makes sense.
How do I make sure the suggestions do the same thing ? Because some times I have to make sure it's in the right position such as when I click on a post.

Reply:
Do you mean the espresso suggestions? My best guess is that you have to do them manually every time. Espresso might have more options that I don't know of tho.

Reply:
It passes the automatic tests, so I think it's better to leave it like that for now